### ISSUE TITLE
[FR] smart lists: show albums

### ISSUE THREAD

**Issue by ildar** [moritz-weber/mucke#54]
instead of separate tracks, i.e. group tracks in a list into albums.
E.g. I want to see most played albums in a smart list.

Thanks a lot for this app: I was thrilled when found it in the FDroid. The good smart lists is one mandatory feature for me when using a music player

**Comment by moritz-weber** [comment]
Thanks for the suggestion. It's not entirely clear, to me, how you imagine this feature.
Do you think of a separate type of smart list that basically works with albums instead of songs? Or an extension of the existing smart lists with grouping abilities?

**Comment by ildar** [comment]
No, it's no need to change the list's data, only presentation. I.e. instead
of showing the list of files show the albums.
The albums list is generated this way: for every song add its album but
only if it wasn't added before

**Comment by ildar** [comment]
I just realized that the desired view is already implemented: it is the
Last Played section of Home.
Can we have (some) smart playlists shown the same way?

**Comment by moritz-weber** [comment]
Going through some open issues, I thought about this one a bit and I think, there is a bit more to it than just displaying the albums instead of songs. This could be easily done, but it would probably make it unintuitive for most users.

These are my semi-structured thoughts on this:
- A smartlist has a quick play button that starts playback with the desired shuffle mode. How would this be implemented here? The existing modes should work, but a "shuffle albums" mode (i.e. shuffle the albums, but keep the songs of the albums in order) would probably be the most intuitive mode.
- There needs to be a new method to generate the queue for this type of smartlist. Just using the songs selected by the smartlist wouldn't work.
- To implement your example (most played albums), one would need to aggregate the play counts of songs in an album (or find another metric that is album-based instead of song-based). This suggests a whole new type of smartlist for albums. Similarly, ordering by number of likes.
- The visualization of this type of smartlist could be very similar or even identical to the usual smart-/playlist, but the logic of selecting albums is probably entirely different. Maybe, other filters and sorting criteria would make sense here.

**Comment by ildar** [comment]
Oh, I'm glad this FR is touched!
I'm mostly agree with your view on this.

On Sun, Aug 27, 2023 at 8:56 AM Moritz Weber wrote:

>
>    - To implement your example (most played albums), one would need to
>    aggregate the play counts of songs in an album (or find another metric that
>    is album-based instead of song-based). This suggests a whole new type of
>    smartlist for albums. Similarly, ordering by number of likes.
>
> No, I think it's just better in simple way: store as before, show
different. Even if it's not accurate from some POV, it's ok.

>
>    - The visualization of this type of smartlist could be very similar or
>    even identical to the usual smart-/playlist, but the logic of selecting
>    albums is probably entirely different. Maybe, other filters and sorting
>    criteria would make sense here.
>
> Maybe yes, maybe not. I'd prefer just quickly find an album among the
recent or most played. Then selecting the album makes the whole album play.
Having it very accurate is overkill.

Thanks a lot!!!
